src/tcp/tcp.ts

```diff
@@ -295,6 +295,14 @@ module Tcp {
               if(this.state_ !== Connection.State.CLOSED) {
                 this.state_ = Connection.State.CONNECTED;
               }
+            })
+            .catch((err) => {
+              log.error('Failed to connect to %1, err=%2',
+                        [JSON.stringify(connectionKind.endpoint),
+                         JSON.stringify(err)]);
+              this.dataToSocketQueue.stopHandling();
```

Comment:
I don't think you need to do this since we only set a handler once `onceConnected` fulfills; and a promise can't both fulfill and reject.